UIforETW/Utility.cpp

```diff
@@ -278,59 +482,64 @@ void SmartEnableWindow(HWND Win, BOOL Enable)
 
 std::wstring GetFilePart(const std::wstring& path)
 {
-	const wchar_t* pLastSlash = wcsrchr(path.c_str(), '\\');
-	if (pLastSlash)
-		return pLastSlash + 1;
+	UIETWASSERT(path.size() > 0);
+	const size_t lastSlash = path.find_last_of( L'\\' );
+	if (lastSlash != std::wstring::npos)
+		return path.substr(lastSlash);
 	// If there's no slash then the file part is the entire string.
 	return path;
 }
 
 std::wstring GetFileExt(const std::wstring& path)
 {
-	std::wstring filePart = GetFilePart(path);
-	const wchar_t* pLastPeriod = wcsrchr(filePart.c_str(), '.');
-	if (pLastPeriod)
-		return pLastPeriod;
+	const std::wstring filePart = GetFilePart(path);
+	const size_t lastPeriod = filePart.find_last_of( L'.' );
+	if (lastPeriod != std::wstring::npos)
+		return filePart.substr(lastPeriod);
+	// If there's no period then there's no extension.
 	return L"";
 }
 
 std::wstring GetDirPart(const std::wstring& path)
 {
-	const wchar_t* pLastSlash = wcsrchr(path.c_str(), '\\');
-	if (pLastSlash)
-		return path.substr(0, pLastSlash + 1 - path.c_str());
+	UIETWASSERT(path.size() > 0);
+	const size_t lastSlash = path.find_last_of( L'\\' );
+	if (lastSlash != std::wstring::npos)
+	{
+		UIETWASSERT(path.at( lastSlash ) != path.back());

```

Comment:
This assert confuses me. First of all, the use of path.at()? Why?

Second, path.at( lastSlash ) will be equal to '\\'. So you're just checking to see if the last character is a '\\'. But that's not illegal, so it's not a valid assert.